From: "K. Y. Srinivasan" <kys@microsoft.com>
Date: Fri, 29 Mar 2013 14:30:38 -0700
Subject: Drivers: hv: vmbus: Fix a bug in hv_need_to_signal()
Patch-mainline: submitted
References: bnc#811417

As part of updating the vmbus protocol, the function hv_need_to_signal()
was introduced. This functions helps optimize signalling from guest to
host. The newly added memory barrier is needed to ensure that we
correctly decide when to signal the host.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Reviewed-by: Haiyang Zhang <haiyangz@microsoft.com>
Reported-by: Olaf Hering <olh@suse.de>

Cc: Stable <stable@vger.kernel.org>  (V3.8+)

Acked-by: <ohering@suse.de>
---
 drivers/hv/ring_buffer.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-3.0-SLE11-SP3/drivers/hv/ring_buffer.c
===================================================================
--- linux-3.0-SLE11-SP3.orig/drivers/hv/ring_buffer.c
+++ linux-3.0-SLE11-SP3/drivers/hv/ring_buffer.c
@@ -71,6 +71,7 @@ u32 hv_end_read(struct hv_ring_buffer_in
 
 static bool hv_need_to_signal(u32 old_write, struct hv_ring_buffer_info *rbi)
 {
+	smp_mb();
 	if (rbi->ring_buffer->interrupt_mask)
 		return false;
 
